packages/std/src/math/decimal.rs

```diff
@@ -224,6 +224,7 @@ impl ops::Mul for Decimal {
             self.numerator().full_mul(other.numerator()) / Self::DECIMAL_FRACTIONAL_UINT256;
         let result: Uint128 = result_as_uint256
             .try_into()
+            .ok()
```

Comment:
Why was this added? Why convert `Result` to `Option` here?

Reply:
I changed this. Initially, I wanted a simpler panic msg here so that the overflow test could expect something simple (`attempt to subtract with overflow`), so I discarded the source error before calling `expect`. Then clippy complained about this, so I just changed the test to not expect any specific msg.

Reply:
What was it complaining about? suspicious_arithmetic_impl?

I think we can bring back the

```
#[should_panic(expected = "attempt to multiply with overflow")]
```

now, can't we?

Reply:
> What was it complaining about? suspicious_arithmetic_impl?

```
warning: called `ok().expect()` on a `Result` value
...
    = note: `#[warn(clippy::ok_expect)]` on by default
```

> I think we can bring back the `#[should_panic(expected = "attempt to multiply with overflow")]` now, can't we?

The error we get is a bit longer than that. It includes the error from the `Result` serialized to a string.

```
      panic message: `"attempt to multiply with overflow: ConversionOverflowError { source_type: \"Uint256\", target_type: \"Uint128\", value: \"343685190590147848098008353506085893569\" }"`,
 expected substring: `"attempt to add with overflow"`
```

I feel a bit weird about hardcoding all of that in the test.

Reply:
Ahh, I see. You want to remove the error details at runtime.

```
#[should_panic(expected = "attempt to multiply with overflow")]
```

only checked if the content is contained, so the tests should pass with that.

Reply:
Let's just go manual with `match` and `panic!`. This way we control the error message and avoid the intermediate `Option` that confised clippy and myself.